src/compiler/types.ts

```diff
@@ -1046,7 +1046,7 @@ namespace ts {
     // @kind(SyntaxKind.JsxOpeningElement)
     export interface JsxOpeningElement extends Expression {
         _openingElementBrand?: any;
-        tagName: EntityName;
+        tagName: LeftHandSideExpression;

```

Comment:
I think I'd rather this have its own type (`type JsxTagNameExpression = Identifier | PropertyAccessExpression | ThisKeyword``) since `LeftHandSideExpression` includes some other subtypes that can't actually appear here (e.g. `CallExpression`)